packages/firestore/src/remote/remote_store.ts

```diff
@@ -228,15 +228,26 @@ export class RemoteStore {
         this.startWatchStream();
       }
 
-      this.updateAndBroadcastOnlineState(OnlineState.Unknown);
+      this.setOnlineStateToUnknown();
 
       return this.fillWritePipeline(); // This may start the writeStream.
     });
   }
 
-  /** Temporarily disables the network. The network can be re-enabled using enableNetwork(). */
-  disableNetwork(): Promise<void> {
-    this.updateAndBroadcastOnlineState(OnlineState.Failed);
+  /**
+   * Temporarily disables the network. The network can be re-enabled using
+   * enableNetwork(). By default this will also update (and broadcast) our
+   * onlineState to OnlineState.Failed. If this should not happen (e.g. because
+   * this is being called as part of shutdown logic or because we're going to
+   * immediately re-enable the network), pass `true` for
+   * suppressOnlineStateFailure.
+   */
+  disableNetwork(suppressOnlineStateFailure?: boolean): Promise<void> {
```

Comment:
Is there any reason why any external caller would ever want to pass `true` for this? Is there any reason to allow it?

If not, consider rearranging this to keep the API as narrow as possible, i.e. by making a private method that takes the intended `OnlineState` as a parameter. This boolean in the public surface is a bit of a wart.

Reply:
I reworked this per your comment here and your out-of-band feedback, see https://github.com/firebase/firebase-js-sdk/pull/358/commits/a9e62e34fe730664869a7a402d76b347a2695f46

Let me know what you think.

Reply:
Looks much better to me. Thank you for taking the time.